### ISSUE TITLE
FlyWay causes SQLException on live reload when migrate-at-start is enabled

### ISSUE THREAD

**Issue by belu** [quarkusio/quarkus#14590]
**Describe the bug**
If FlyWay is configured with enabled `migrate-at-start`, this causes an SQLException during live-reload.

**Expected behavior**
No exception should occure.

**Actual behavior**
An exception happens during the restart of the application:
```
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-01-25 20:29:42,968 INFO  [org.fly.cor.int.lic.VersionPrinter] (Quarkus Main Thread) Flyway Community Edition 7.4.0 by Redgate
2021-01-25 20:29:43,033 INFO  [org.fly.cor.int.dat.bas.DatabaseType] (Quarkus Main Thread) Database: jdbc:h2:mem:test (H2 1.4)
2021-01-25 20:29:43,098 INFO  [org.fly.cor.int.com.DbClean] (Quarkus Main Thread) Successfully dropped pre-schema database level objects (execution time 00:00.000s)
2021-01-25 20:29:43,101 INFO  [org.fly.cor.int.com.DbClean] (Quarkus Main Thread) Successfully cleaned schema "PUBLIC" (execution time 00:00.002s)
2021-01-25 20:29:43,102 INFO  [org.fly.cor.int.com.DbClean] (Quarkus Main Thread) Successfully dropped post-schema database level objects (execution time 00:00.000s)
2021-01-25 20:29:43,104 INFO  [org.fly.cor.int.lic.VersionPrinter] (Quarkus Main Thread) Flyway Community Edition 7.4.0 by Redgate
2021-01-25 20:29:43,130 INFO  [org.fly.cor.int.sch.JdbcTableSchemaHistory] (Quarkus Main Thread) Creating Schema History table "PUBLIC"."flyway_schema_history" ...
2021-01-25 20:29:43,176 INFO  [org.fly.cor.int.com.DbMigrate] (Quarkus Main Thread) Current version of schema "PUBLIC": << Empty Schema >>
2021-01-25 20:29:43,179 INFO  [org.fly.cor.int.com.DbMigrate] (Quarkus Main Thread) Schema "PUBLIC" is up to date. No migration necessary.
2021-01-25 20:29:43,298 INFO  [io.quarkus] (Quarkus Main Thread) quarkus-flyway-sqlexception 1.0.0-SNAPSHOT on JVM (powered by Quarkus 1.11.0.Final) started in 1.990s. Listening on: http://localhost:8080
2021-01-25 20:29:43,300 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-01-25 20:29:43,300 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [agroal, cdi, flyway, jdbc-h2, mutiny, narayana-jta, resteasy, smallrye-context-propagation]
2021-01-25 20:29:55,003 INFO  [io.qua.dep.dev.RuntimeUpdatesProcessor] (vert.x-worker-thread-1) File change detected: /home/belu/git/quarkus-flyway-sqlexception/src/main/resources/application.properties
2021-01-25 20:29:55,041 INFO  [io.quarkus] (Quarkus Main Thread) quarkus-flyway-sqlexception stopped in 0.036s
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2021-01-25 20:29:55,348 INFO  [org.fly.cor.int.lic.VersionPrinter] (Quarkus Main Thread) Flyway Community Edition 7.4.0 by Redgate
2021-01-25 20:29:55,350 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.sql.SQLException: This pool is closed and does not handle any more connections!

2021-01-25 20:29:55,353 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (vert.x-worker-thread-1) Failed to start quarkus: java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus

        at io.quarkus.runner.bootstrap.StartupActionImpl.runMainClass(StartupActionImpl.java:145)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.restartApp(IsolatedDevModeMain.java:190)
        at io.quarkus.deployment.dev.IsolatedDevModeMain.restartCallback(IsolatedDevModeMain.java:174)
        at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.doScan(RuntimeUpdatesProcessor.java:237)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$2.handle(VertxHttpHotReplacementSetup.java:62)
        at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup$2.handle(VertxHttpHotReplacementSetup.java:52)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:787)
        at io.quarkus.runtime.Application.start(Application.java:90)
        at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:97)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:66)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:119)
        at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$3.run(StartupActionImpl.java:134)
        ... 1 more
Caused by: org.flywaydb.core.internal.exception.FlywaySqlException: 
Unable to obtain connection from database: This pool is closed and does not handle any more connections!
--------------------------------------------------------------------------------------------------------
SQL State  : null
Error Code : 0
Message    : This pool is closed and does not handle any more connections!

        at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:68)
        at org.flywaydb.core.internal.jdbc.JdbcConnectionFactory.<init>(JdbcConnectionFactory.java:69)
        at org.flywaydb.core.Flyway.execute(Flyway.java:507)
        at org.flywaydb.core.Flyway.clean(Flyway.java:358)
        at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:58)
        at io.quarkus.deployment.steps.FlywayProcessor$createBeansAndStartActions-1520831253.deploy_0(FlywayProcessor$createBeansAndStartActions-1520831253.zig:80)
        at io.quarkus.deployment.steps.FlywayProcessor$createBeansAndStartActions-1520831253.deploy(FlywayProcessor$createBeansAndStartActions-1520831253.zig:40)
        at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:600)
        ... 12 more
Caused by: java.sql.SQLException: This pool is closed and does not handle any more connections!
        at io.agroal.pool.ConnectionPool.beforeAcquire(ConnectionPool.java:199)
        at io.agroal.pool.ConnectionPool.getConnection(ConnectionPool.java:206)
        at io.agroal.pool.DataSource.getConnection(DataSource.java:81)
        at org.flywaydb.core.internal.jdbc.JdbcUtils.openConnection(JdbcUtils.java:59)
        ... 19 more
```

**To Reproduce**

Clone the following project: https://github.com/belu/quarkus-flyway-sqlexception

1. Start Quarkus: `mvn clean compile quarkus:dev`
2. Open URL: http://localhost:8080/hello-resteasy
3. Change `application.properties` (e. g. add a new empty line and save file)
4. Open URL again: http://localhost:8080/hello-resteasy. This triggers a restart of the Quarkus app which causes the Exception.

**Configuration**
```application.properties

quarkus.datasource.db-kind=h2
quarkus.datasource.jdbc.url=jdbc:h2:mem:test

quarkus.flyway.migrate-at-start=true
```
For the test repo a H2 in-memory database is used. However, this issue also occured with a configured PostgreSQL.

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
`Linux linux-p74w 4.12.14-lp151.28.67-default #1 SMP Fri Sep 4 15:23:21 UTC 2020 (2c5a14f) x86_64 x86_64 x86_64 GNU/Linux`

 - Output of `java -version`: 
```
openjdk version "11.0.8" 2020-07-14
OpenJDK Runtime Environment (build 11.0.8+10-suse-lp151.3.19.1-x8664)
OpenJDK 64-Bit Server VM (build 11.0.8+10-suse-lp151.3.19.1-x8664, mixed mode)
```

 - Quarkus version or git rev: 
`1.11.0.Final`

 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
`Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)`

**Comment by belu** [comment]
Further info: this bug does not occure with `1.10.5.Final`.

Was this introduced by one of the following changes?
- https://github.com/quarkusio/quarkus/pull/14085
- https://github.com/quarkusio/quarkus/issues/11409

**Comment by gsmet** [comment]
It will be fixed in 1.11.1.Final released on Wednesday.

**Comment by gsmet** [comment]
For the record, this is the associated PR: https://github.com/quarkusio/quarkus/pull/14372 .
